docs/concepts/node-agent.md

```diff
@@ -32,6 +32,14 @@ It also executes tasks that need to be executed before any other components are
 
 This section describes the controllers in more details.
 
+### [`Lease` Controller](../../pkg/nodeagent/controller/lease)
+
+This controller creates a lease for `gardener-node-agent` in `kube-system` namespace of the shoot cluster.
+Each instance of `gardener-node-agent` creates its own lease when its corresponding `Node` was created.
+It renews the `Lease` resource every 10 seconds. This indicates a heartbeat to the external world.
+
+At the moment, the `Lease` resources are not reflected in any health check function of the shoot cluster yet.
```

Comment:
I would remove this sentence - we might easily forget about it, and we plan to work on this anyways soon. Also, this document should rather focus on what the controller does and why, but not how others use it :)